src/SensorReading.cpp

```diff
@@ -36,19 +36,26 @@ std::map<fault_index_type, uint8_t> map_to_mask = {
     {fault_index_type::burn_wire, constants::fault::burn_wire},
     {fault_index_type::sd_card, constants::fault::sd_card},
     {fault_index_type::camera_on_failed, constants::fault::camera_on_failed},
-    {fault_index_type::light_val, constants::fault::light_val},
 }; // map from a fault type to the mask
 
 SensorReading::SensorReading(fault_index_type t, float x, bool fault)
```

Comment:
SensorReading constructor should allow us to specify the size of the buffer.

Reply:
constructor
-> min and max acceptable value

getValid (now private)
-> if (buffer is empty or valid == False) return False

setInvalid
-> clear buffer 

setValue
-> check if SensorReading is in range of min and max valid values
-> if it is range add to buffer

boolean getValue( uint8 * v)
{
if(getValid())
*v = value;
return 1

else
return 0
}

Example of how getValue is called:
uint32 gyro_x;
if ( getValue( (uint8 *)gyro_x) )
{
    do stuff with gyro_x
} else {
    do nothing
}

Reply:
all non zero values are true- return 0